### PR TITLE
Silabs CPMS support added.

### PR DESCRIPTION
**Pull request by rcasallas-silabs**
Added support for Silicon Labs CPMS data provisioning.

SilabsDeviceAttestationCreds now support both header-provided sizes, and offsets, or size and offsets stored in NVM3. This will allow to provision the attestation data with either the old `creds.py` script, or the new CPMS `provision.py` script.

Tested on EFR32MG12 and EFR32MG24.

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]

File: examples/platform/silabs/SilabsDeviceAttestationCreds.cpp [hunk 1]
```diff
@@ -33,12 +37,26 @@ namespace {
...
+           SilabsConfig::ConfigValueExists(SilabsConfig::kConfigKey_Creds_CD_Size))
+        {
+            ReturnErrorOnFailure(SilabsConfig::ReadConfigValue(SilabsConfig::kConfigKey_Creds_CD_Offset, offset));
+            ReturnErrorOnFailure(SilabsConfig::ReadConfigValue(SilabsConfig::kConfigKey_Creds_CD_Size, size));
+        }
+
+        uint8_t * address = _credentials_address + offset;
+        ByteSpan cd_span(address, size);
+        ChipLogProgress(DeviceLayer, "GetCertificationDeclaration, addr:%p, size:%lu", address, size);
+        ChipLogByteSpan(Zcl, ByteSpan(cd_span.data(), kDebugLength));
```
Comment (bzbarsky-apple) [inline examples/platform/silabs/SilabsDeviceAttestationCreds.cpp:58]: Are we guaranteed that `kDebugLength < size` here?  Might be worth documenting why, if we are.  Same in the other places like this below.

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/platform/silabs/SilabsDeviceAttestationCreds.cpp (hunk 1 above)
Comment (rcasallas-silabs) [reply examples/platform/silabs/SilabsDeviceAttestationCreds.cpp:58]: Miss-aligned certificates is a common error, and printing the first few bytes is quite useful to validate this. Eight is just enough for this purpose. I will add this comment into the code.

Comment (bzbarsky-apple) [reply examples/platform/silabs/SilabsDeviceAttestationCreds.cpp]: My point is that if `kDebugLength > size` then we will be printing random data.  But presumably `size` is bigger than `kDebugLength` in practice.

Comment (rcasallas-silabs) [reply examples/platform/silabs/SilabsDeviceAttestationCreds.cpp]: Size is less than 8 only if the files are missing, or corrupted. I have added a check to cover this cases.

**Pull request merged**